include/powsybl/stdcxx/reference.hpp

```diff
@@ -129,14 +129,19 @@ Reference<T> ref(T& reference) {
 
 template <typename T>
 Reference<T> ref(const std::unique_ptr<T>& pointer) {
-    return Reference<T>(*pointer);
+    return static_cast<bool>(pointer) ? Reference<T>(*pointer) : Reference<T>();
 }
 
 template <typename T, typename U, typename = typename std::enable_if<std::is_base_of<T, U>::value && !std::is_same<T, U>::value>::type>
 Reference<T> ref(U& reference) {
     return Reference<T>(dynamic_cast<T&>(reference));
 }
 
+template <typename T>
+Reference<T> ref(const Reference<T>& reference) {
+    return Reference<T>(reference);
+}
+
```

Comment:
I had to add this one because the following code crashed : 
```
    Reference<int> myRef;

    Reference<int> refFromInvalidRef = ref<int>(myRef);
    BOOST_CHECK(!refFromInvalidRef);
    ```

Explanation: because of the operator
```
    Reference::operator T&() const {
        return get();
    }
```

the following function is called
```
template <typename T>
Reference<T> ref(T& reference) {
    return Reference<T>(reference);
}
```